### PR TITLE
Split up cmp_owned tests, add run-rustfix

### PR DESCRIPTION
**Pull request by phansch**
Some of the cmp_owned tests emitted non-machine-applicable suggestions,
so I moved them to `tests/ui/cmp_owned/without_suggestion.rs` and added
`// run-rustfix` to the other half.

changelog: none

cc #3630

### CODE REVIEW

File: tests/ui/cmp_owned/with_suggestion.rs [hunk 1]
```diff
@@ -22,21 +24,15 @@ fn main() {
...
-
-    y.to_owned() == **x;
 }
 
 struct Foo;
 
 impl PartialEq for Foo {
+    // Allow this here, because it emits the lint
+    // without a suggestion. This is tested in
+    // `tests/ui/cmp_owned_without_suggestion.rs`
```
Comment (flip1995) [inline tests/ui/cmp_owned/with_suggestion.rs]: NIT:
```suggestion
    // `tests/ui/cmp_owned/without_suggestion.rs`
```

r=me after that.

File: tests/ui/cmp_owned/with_suggestion.fixed [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    "abc".chars().filter(|c| *c != 'X');
+}
+
+struct Foo;
+
+impl PartialEq for Foo {
+    // Allow this here, because it emits the lint
+    // without a suggestion. This is tested in
+    // `tests/ui/cmp_owned_without_suggestion.rs`
```
Comment (phansch) [inline tests/ui/cmp_owned/with_suggestion.fixed]:
```suggestion
    // `tests/ui/cmp_owned/without_suggestion.rs`
```

**Comment by flip1995** [comment]
Thanks!

@bors r+

**Comment by phansch** [comment]
@flip1995 thanks for doing all those reviews!

**Pull request merged**